src/Transaction.js

```diff
@@ -23,27 +18,27 @@ Transaction.prototype = {
   done   : false,
   error  : null,
 
-  finish(error, body) {
-    // This is a neat trick to get around the promise try/catch
-    // https://github.com/then/promise/blob/master/src/done.js
-    setTimeout(this.callback.bind(this, error, body), 0)
-  },
-
   isComplete() {
     return this.error || this.done
   },
 
-  run() {
+  run(resolve, reject, progress) {
```

Comment:
I wanted this to follow the signature of Promises (using the old progress callback from the A specification)